client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,170 @@
+import { gql, useMutation } from "@apollo/client";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import Form from "react-bootstrap/Form";
+import moment from "moment";
+import Table from "react-bootstrap/Table";
+
+import Request from "../../data/types/request";
+import RequestForm from "../organisms/RequestForm";
+
+interface Props {
+  requests: Request[];
+  onChangeRequests: (requests: Request[]) => void;
+}
+
+const ClientRequestTable: FunctionComponent<Props> = (props: Props) => {
+  const [requestSelectedForEditing, setRequestSelectedForEditing] =
+    useState("");
+  const [requests, setRequests] = useState<Request[]>([]);
+
+  const headingList = [
+    "Fulfilled",
+    "Request Group",
+    "Request Type",
+    "Quantity",
+    "Date Requested",
+    "",
+  ];
+  const updateRequest = gql`
+    mutation updateRequest($request: RequestInput) {
+      updateRequest(request: $request) {
+        id
+        success
+        message
+      }
+    }
+  `;
+  const deleteRequest = gql`
+    mutation deleteRequest($id: ID) {
+      softDeleteRequest(id: $id) {
+        id
+        success
+        message
+      }
+    }
+  `;
+  const [mutateDeleteRequest] = useMutation(deleteRequest);
+  const [mutateRequest] = useMutation(updateRequest);
+
+  const sortRequests = (unsortedRequests: Request[]): Request[] => {
+    const requests = unsortedRequests
+      .filter((request) => !request.deleted)
+      .sort((a, b) => {
+        return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+      });
+
+    // separate fulfilled and non-fulfilled requests
+    const nonFulfilledRequests = requests.filter(
+      (request) => !request.fulfilled
+    );
+    const fulfilledRequests = requests.filter((request) => request.fulfilled);
+
+    const sortedRequests: Request[] = nonFulfilledRequests!.concat(
+      fulfilledRequests!
```

Comment:
If `fulfilledRequests` is `null` or `undefined`, we'd want to concat and empty array `[]`